Cython/Compiler/Pipeline.py

```diff
@@ -130,6 +182,8 @@ def inject_utility_code_stage(module_node):
                 module_node.merge_in(tree.with_compiler_directives(),
                                      tree.scope, stage="utility",
                                      merge_scope=True)
+            elif hasattr(utilcode, 'pxd_scope') and utilcode.pxd_scope:
```

Comment:
1) I'd avoid looking up the attribute twice.
```suggestion
            elif getattr(utilcode, 'pxd_scope', None):
```
2) Rather than allowing the attribute to be optional, why not make sure it's always available (and `None` if unused)?